### DISCUSSION TITLE
GameActivity, location, events

### DISCUSSION THREAD

**Discussion by timprepscius** [android/ndk#1932] [Q&A]
Is there anything with the NDK looper + GameActivity that could cause location events to be dropped?

I am using the NDK with a GameActivity.
I'm using HelloVK as the starting point.

I have incorporated location updates by:
1. checking permissions
2. requesting via:
```
                    locationManager.requestLocationUpdates(
                        provider,
                        minimumTimeMS,
                        minimumDistanceM,
                        locationListener
                    )

```

where provide is GPS given by ```locationManager.getBestProvider(criteria, true);```

locationManager is a ```locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager```
user permission for FINE and COARSE locations are in the manifest.  Everything works.  No exceptions thrown.

Except-> no location updates are passed into the location locationListener.

I can see in the logcat things like:

```
09-15 09:11:06.920   912  3029 V LocSvc_HIDL_IzatProvider: [onLocationChanged][67] [HS] =>>>> [HC]
09-15 09:11:06.920  2014  2606 V LocSvc_HIDL_IzatProvider_jni: [onLocationChanged][136] [HC] <<= [HS]
09-15 09:11:06.921   912  3029 V LocSvc_HIDL_IzatProvider: [onLocationChanged][67] [HS] =>>>> [HC]
09-15 09:11:06.922  2014  2328 V LocSvc_HIDL_IzatProvider_jni: [onLocationChanged][136] [HC] <<= [HS]
```

and I can see when it is activated:
```
09-15 09:10:37.374   912  3029 V LocSvc_HIDL_IzatProvider: [onStatusChanged][86] [HS] =>>>> [HC]
09-15 09:10:37.374  2014  2328 V LocSvc_HIDL_IzatProvider_jni: [onStatusChanged][185] [HC] <<= [HS]
```

Is there some peculiarity with the looper and events and GameActivity that might cause those events to be "dropped" or "consumed" before they get to the LocationListener?

The game activity loop looks like this:

```
void android_main(struct android_app *state) {
  VulkanEngine engine{};
  vkt::HelloVK vulkanBackend{};

  engine.app = state;
  engine.app_backend = &vulkanBackend;
  state->userData = &engine;
  state->onAppCmd = HandleCmd;

  ALooper_prepare(ALOOPER_PREPARE_ALLOW_NON_CALLBACKS);

  while (true) {
    int ident;
    int events;
    android_poll_source *source;

    while (!engine.canRender) {
      ALooper_pollAll(
          -1, 
          nullptr, 
          &events,
          (void **)&source);

      if (source != nullptr)
          source->process(state, source);
    }

    ALooper_pollAll(
      0, 
      nullptr, 
      &events,
      (void **)&source
    );

    if (source != nullptr)
        source->process(state, source);

    engine.attitude.Process();
    HandleSensorEvents(state);
    HandleInputEvents(state);

    if (engine.canRender)
      engine.app_backend->render();
  }
}
```

Thanks.

Yes I know this might very well not be an NDK problem, but I want to check - because everything seems fine.

**Comment by DanAlbert** [comment]
> Yes I know this might very well not be an NDK problem

Yeah, we don't know. You'd need to ask the folks that made `GameActivity`, or maybe the folks that made `ALooper`.

https://issuetracker.google.com/issues?q=status:open%20componentid:897320 looks to be the buganizer component for the games sdk. For `Looper` you want the "Framework" component on http://b.android.com (the games sdk really should be listed there too but isn't, I'll go fix that for them). I'd recommend starting with the games folks and they can move it if it's the wrong place.

**Comment by timprepscius** [comment]
I'm going to post, just in case someone searches, preferably using a search service which doesn't track you, like duck duck go.

Don't use locationManager = getSystemService(Context.LOCATION_SERVICE)

It doesn't work.  
Use the fused location service.  Then things will work.
